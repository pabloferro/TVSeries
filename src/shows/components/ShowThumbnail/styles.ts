import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    paddingHorizontal: 20,
    paddingVertical: 10,
    marginBottom: 5,
  },
  infoContainer: {
    padding: 10,
    paddingVertical: 10,
  },
  poster: {
    width: 71,
    height: 100,
  },
});
