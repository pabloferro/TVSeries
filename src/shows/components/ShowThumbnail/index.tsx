import React from 'react';
import {View} from 'react-native';
import FastImage from 'react-native-fast-image';
import CustomText from '../../../components/CustomText';
import {TextVariants} from '../../../components/CustomText/TextVariants';
import ServerImage from '../../../components/ServerImage';
import {Show} from '../../api/Show';

import styles from './styles';

export default function ShowThumbnail({show}: {show: Show}) {
  return (
    <View style={styles.container}>
      {show.image && (
        <ServerImage
          style={styles.poster}
          source={{
            uri: show.image.medium,
          }}
          resizeMode={FastImage.resizeMode.contain}
        />
      )}
      <View style={styles.infoContainer}>
        <CustomText
          variant={TextVariants.h2}
          ellipsizeMode="tail"
          numberOfLines={2}>
          {show.name}
        </CustomText>
        <CustomText variant={TextVariants.secondary}>
          ({new Date(show.premiered).getFullYear()})
        </CustomText>
        <CustomText variant={TextVariants.body}>
          Rating: {show.rating.average}
        </CustomText>
      </View>
    </View>
  );
}
