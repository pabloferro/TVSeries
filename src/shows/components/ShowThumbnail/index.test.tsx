import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';

import {mockedNavigate} from '../../../../jest/setup';
import {EXAMPLE_SHOW} from './__test__/exampleShow';

import ShowThumbnail from '.';

jest.mock(
  '../../../favorites/components/FavoriteIndicator',
  () => 'MockedFavoriteIndicator',
);

test('renders correctly with rating', () => {
  const {toJSON} = render(<ShowThumbnail show={EXAMPLE_SHOW} />);
  expect(toJSON()).toMatchSnapshot();
});

test('renders correctly without rating', () => {
  const {toJSON} = render(
    <ShowThumbnail show={{...EXAMPLE_SHOW, rating: undefined}} />,
  );
  expect(toJSON()).toMatchSnapshot();
});

test('displays the name of the show', () => {
  const {getByText} = render(<ShowThumbnail show={EXAMPLE_SHOW} />);

  expect(getByText(EXAMPLE_SHOW.name)).toBeTruthy();
});

test('displays the poster of the show', () => {
  const {getByA11yLabel} = render(<ShowThumbnail show={EXAMPLE_SHOW} />);

  const poster = getByA11yLabel(`Poster of ${EXAMPLE_SHOW.name}`);

  expect(poster.props.source).toEqual({uri: EXAMPLE_SHOW.image?.medium});
});

test('navigates to ShowDetailScreen on press', () => {
  const {getByA11yRole} = render(<ShowThumbnail show={EXAMPLE_SHOW} />);

  const thumbnail = getByA11yRole('button');
  fireEvent.press(thumbnail);

  expect(mockedNavigate).toHaveBeenCalledWith('ShowDetail', {
    show: EXAMPLE_SHOW,
  });
});
