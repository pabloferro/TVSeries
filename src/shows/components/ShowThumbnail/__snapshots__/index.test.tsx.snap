// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with rating 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 5,
      "opacity": 1,
      "paddingVertical": 10,
    }
  }
>
  <View
    style={
      Object {
        "height": 100,
        "width": 71,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "height": "100%",
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "backgroundColor": "#555",
          },
        ]
      }
    />
    <View
      collapsable={false}
      style={
        Object {
          "height": "100%",
          "opacity": 0,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "height": "100%",
              "position": "absolute",
              "width": "100%",
            },
          ]
        }
      >
        <FastImageView
          accessibilityLabel="Poster of Under the Dome"
          onFastImageLoadEnd={[Function]}
          resizeMode="contain"
          source={
            Object {
              "uri": "https://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#FFF",
              "fontSize": 18,
              "fontWeight": "500",
            },
            undefined,
          ]
        }
      >
        Under the Dome
      </Text>
      <MockedFavoriteIndicator
        id={1}
        type="show"
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#999",
            "fontSize": 16,
            "fontWeight": "500",
          },
          undefined,
        ]
      }
    >
      (
      2013
      )
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 16,
            "fontWeight": "400",
          },
          undefined,
        ]
      }
    >
      Rating: 
      6.5
    </Text>
  </View>
</View>
`;

exports[`renders correctly without rating 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 5,
      "opacity": 1,
      "paddingVertical": 10,
    }
  }
>
  <View
    style={
      Object {
        "height": 100,
        "width": 71,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "height": "100%",
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "backgroundColor": "#555",
          },
        ]
      }
    />
    <View
      collapsable={false}
      style={
        Object {
          "height": "100%",
          "opacity": 0,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "height": "100%",
              "position": "absolute",
              "width": "100%",
            },
          ]
        }
      >
        <FastImageView
          accessibilityLabel="Poster of Under the Dome"
          onFastImageLoadEnd={[Function]}
          resizeMode="contain"
          source={
            Object {
              "uri": "https://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#FFF",
              "fontSize": 18,
              "fontWeight": "500",
            },
            undefined,
          ]
        }
      >
        Under the Dome
      </Text>
      <MockedFavoriteIndicator
        id={1}
        type="show"
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#999",
            "fontSize": 16,
            "fontWeight": "500",
          },
          undefined,
        ]
      }
    >
      (
      2013
      )
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 16,
            "fontWeight": "400",
          },
          undefined,
        ]
      }
    >
      Rating: 
      -
    </Text>
  </View>
</View>
`;
